tests/conftest.py

```diff
@@ -572,19 +577,21 @@ async def mqtt_mock_entry(hass, mqtt_client_mock, mqtt_config):
     real_mqtt_instance = None
     mock_mqtt_instance = None
 
-    async def _setup_mqtt_entry():
+    async def _setup_mqtt_entry(setup_entry):
         """Set up the MQTT config entry."""
-        if entry.state == ConfigEntryState.NOT_LOADED:
+        if setup_entry:
```

Comment:
Pass a function to call instead if we want to reuse code. The point with my comment was to avoid this condition completely.

Reply:
I agree with Erik, this is not helping. I suggest reverting this last commit.

Reply:
I don't think we need to revert it now when it's already implemented the way @MartinHjelmare suggested 👍

Reply:
It does matter. We should not have branching in tests.